frontend/src/components/Room.tsx

```diff
@@ -94,20 +103,67 @@ function Room() {
     } ));
 
 
+    const [roomAnnotation, setRoomAnnotation ] = useState(room?.roomAnnotation===null ? "" : String(room?.roomAnnotation))
+
+    const roomAnnotationChange = (e) => {
+        e.preventDefault();
+        pushNewRoomAnnotation.mutate(e.target.value)
+        setRoomAnnotation(e.target.value)
+    }
+
+    const pushNewRoomAnnotation = useMutation((newAnnotation) => {
```

Comment:
all hooks should be on top of the component, you can just move it before roomAnnotationChange